From db729f3b685fd832a3ec7387b339cf2bbeb4bd4d Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@gmail.com>
Date: Wed, 11 May 2016 21:45:56 -0400
Subject: Update mdds to 1.2.0.  Note that the API version is also up.

liborcus-0.11.2 is out only to make it buildable with mdds-1.2.

Change-Id: I9648d827b008da252c57be0ebfd2efccb008ac70
Reviewed-on: https://gerrit.libreoffice.org/24944
Tested-by: Jenkins <ci@libreoffice.org>
Reviewed-by: Kohei Yoshida <libreoffice@kohei.us>

diff --git a/configure.ac b/configure.ac
index 3f1b7f0..6dea282 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8832,7 +8832,7 @@ AC_SUBST(SYSTEM_BOOST)
 dnl ===================================================================
 dnl Check for system mdds
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([mdds], [MDDS], [mdds >= 0.12.0], ["-I${WORKDIR}/UnpackedTarball/mdds/include"])
+libo_CHECK_SYSTEM_MODULE([mdds], [MDDS], [mdds-1.2 >= 1.2.0], ["-I${WORKDIR}/UnpackedTarball/mdds/include"])
 
 dnl ===================================================================
 dnl Check for system glm
diff --git a/download.lst b/download.lst
index a2483c2..ddd6edc 100644
--- a/download.lst
+++ b/download.lst
@@ -94,8 +94,8 @@
 export LIBXSLT_TARBALL := 9667bf6f9310b957254fdcf6596600b7-libxslt-1.1.28.tar.gz
 export LPSOLVE_TARBALL := 26b3e95ddf3d9c077c480ea45874b3b8-lp_solve_5.5.tar.gz
 export MARIADB_TARBALL := a233181e03d3c307668b4c722d881661-mariadb_client-2.0.0-src.tar.gz
-export MDDS_MD5SUM := ef2560ed5416652a7fe195305b14cebe
-export MDDS_TARBALL := mdds_0.12.1.tar.bz2
+export MDDS_MD5SUM := 9f3383fb7bae825eab69f3a6ec1d74b2
+export MDDS_TARBALL := mdds-1.2.0.tar.bz2
 export MSPUB_MD5SUM := ff9d0f9dd8fbc523408ea1953d5bde41
 export MSPUB_TARBALL := libmspub-0.1.2.tar.bz2
 export MWAW_MD5SUM := 4a8a53a9d997cf0e2bd208178797dbfb
diff --git a/sc/inc/mtvelements.hxx b/sc/inc/mtvelements.hxx
index 0bc5162..18edafc 100644
--- a/sc/inc/mtvelements.hxx
+++ b/sc/inc/mtvelements.hxx
@@ -53,6 +53,7 @@ const mdds::mtv::element_t element_type_cellnote = mdds::mtv::element_type_user_
 /// Mapped standard element types (for convenience).
 const mdds::mtv::element_t element_type_numeric = mdds::mtv::element_type_numeric;
 const mdds::mtv::element_t element_type_empty = mdds::mtv::element_type_empty;
+const mdds::mtv::element_t element_type_uint16 = mdds::mtv::element_type_ushort;
 
 /// Custom element blocks.
 
@@ -65,6 +66,7 @@ typedef mdds::mtv::noncopyable_managed_element_block<element_type_formula, ScFor
 
 /// Mapped standard element blocks (for convenience).
 typedef mdds::mtv::numeric_element_block numeric_block;
+typedef mdds::mtv::ushort_element_block uint16_block;
 
 /// This needs to be in the same namespace as CellTextAttr.
 MDDS_MTV_DEFINE_ELEMENT_CALLBACKS(CellTextAttr, element_type_celltextattr, CellTextAttr(), celltextattr_block)
diff --git a/sc/source/core/tool/scmatrix.cxx b/sc/source/core/tool/scmatrix.cxx
index f1ce9a5..1689809 100644
--- a/sc/source/core/tool/scmatrix.cxx
+++ b/sc/source/core/tool/scmatrix.cxx
@@ -57,17 +57,15 @@ using ::std::unary_function;
  * Custom string trait struct to tell mdds::multi_type_matrix about the
  * custom string type and how to handle blocks storing them.
  */
-struct custom_string_trait
+struct matrix_trait
 {
-    typedef svl::SharedString string_type;
     typedef sc::string_block string_element_block;
-
-    static const mdds::mtv::element_t string_type_identifier = sc::element_type_string;
+    typedef sc::uint16_block integer_element_block;
 
     typedef mdds::mtv::custom_block_func1<sc::string_block> element_block_func;
 };
 
-typedef mdds::multi_type_matrix<custom_string_trait> MatrixImplType;
+typedef mdds::multi_type_matrix<matrix_trait> MatrixImplType;
 
 namespace {
 
@@ -2304,6 +2302,7 @@ public:
                 }
             }
             break;
+            case mdds::mtm::element_integer:
             case mdds::mtm::element_string:
             case mdds::mtm::element_empty:
             default:
diff --git a/svl/source/misc/gridprinter.cxx b/svl/source/misc/gridprinter.cxx
index 4044b3e..70e7439 100644
--- a/svl/source/misc/gridprinter.cxx
+++ b/svl/source/misc/gridprinter.cxx
@@ -26,12 +26,10 @@ const mdds::mtv::element_t element_type_string = mdds::mtv::element_type_user_st
 // String block
 typedef mdds::mtv::default_element_block<element_type_string, OUString> string_block;
 
-struct custom_string_trait
+struct matrix_trait
 {
-    typedef OUString string_type;
     typedef string_block string_element_block;
-
-    static const mdds::mtv::element_t string_type_identifier = element_type_string;
+    typedef mdds::mtv::ushort_element_block integer_element_block;
 
     typedef mdds::mtv::custom_block_func1<string_block> element_block_func;
 };
@@ -48,7 +46,7 @@ MDDS_MTV_DEFINE_ELEMENT_CALLBACKS(OUString, svl::element_type_string, OUString()
 
 namespace svl {
 
-typedef mdds::multi_type_matrix<custom_string_trait> MatrixImplType;
+typedef mdds::multi_type_matrix<matrix_trait> MatrixImplType;
 
 struct GridPrinter::Impl
 {
-- 
cgit v0.10.2

